bzlib.rs

```diff
@@ -11,13 +11,47 @@ use crate::libbzip2_rs_sys_version;
 use crate::BZ_MAX_UNUSED;
 
 // FIXME remove this
+#[cfg(not(target_os = "windows"))]
 extern "C" {
     #[cfg_attr(target_os = "macos", link_name = "__stdinp")]
     static mut stdin: *mut FILE;
     #[cfg_attr(target_os = "macos", link_name = "__stdoutp")]
     static mut stdout: *mut FILE;
 }
 
+#[cfg(all(target_os = "windows", target_env = "gnu"))]
+extern "C" {
+    fn __acrt_iob_func(idx: libc::c_uint) -> *mut FILE;
+}
+
+#[cfg(not(target_os = "windows"))]
+macro_rules! STDIN {
+    () => {
+        stdin
+    };
+}
+
+#[cfg(all(target_os = "windows", target_env = "gnu"))]
+macro_rules! STDIN {
+    () => {
+        __acrt_iob_func(0)
+    };
+}
+
+#[cfg(not(target_os = "windows"))]
+macro_rules! STDOUT {
+    () => {
+        stdout
+    };
+}
+
+#[cfg(all(target_os = "windows", target_env = "gnu"))]
+macro_rules! STDOUT {
+    () => {
+        __acrt_iob_func(1)
+    };
+}
+
```

Comment:
would it work to use `std::io::stdin().as_raw_fd`  and then use the `libc::fdopen`  function on that to get the `*FILE` ?

Reply:
I think not. It would require dup on the fd to avoid closing the FILE from closing stdin/stdout and there are a couple of places where we compare against the specific FILE instance for stdin/stdout.

Reply:
well in this file only `bzopen_or_bzdopen`  uses these, and so I think we can refactor so that the equality check is not a problem at least. For the library I think that' s worth it, for `bzip2.rs` it might be too complicated

Reply:
but we can do that in a separate PR of course

Reply:
I missed another occurence in `BZ2_bzclose`, that complicates things

Reply:
The library also compares against stdin/stdout. If I were to pass `stdin` to the library and the check in the library was either removed or replaced with a check against `fdopen`, it will falsely conclude that it can close `stdin`.